irods/connection.py

```diff
@@ -325,7 +325,15 @@ def disconnect(self):
                     self.socket = self.socket.unwrap()
                 except AttributeError:
                     pass
-                self.socket.shutdown(socket.SHUT_RDWR)
+                try:
+                    # On MacOS this can result in the following error:
+                    # OSError: [Errno 57] Socket is not connected
+                    self.socket.shutdown(socket.SHUT_RDWR)
+                except OSError as e:
+                    import errno
+                    # Re-raise if not the known MacOS error
+                    if e.errno != errno.ENOTCONN:
+                        raise
```

Comment:
Two questions:
1. Should this code be wrapped in an OS check (i.e. if on MacOS)?
2. Are we potentially hiding the ENOTCONN error for other cases?

Reply:
1. mmm, maybe, but i think question 2 is more important
2. since the point of a `.shutdown()` call is to end with a no longer connected socket... i think we're fine.

Reply:
I agree. This is fine.

Reply:
A Python implementation which doesn't have this bug will never trigger this unless the users is doing something extremely weird.

Resolving.